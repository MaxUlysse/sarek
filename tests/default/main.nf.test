nextflow_pipeline {

    name "default"
    script "main.nf"

    test("Should run without failures") {
        when {
            params {
                outdir = baseDir.toURI().relativize(outputDir.toURI()).toString()
            }
        }

        then {
            assert path("$outputDir/multiqc").exists()

            // Removed once snapshots are fixed
            assert path("$outputDir/csv/markduplicates.csv").exists()
            assert path("$outputDir/csv/markduplicates_no_table.csv").exists()
            assert path("$outputDir/csv/recalibrated.csv").exists()
            assert path("$outputDir/csv/variantcalled.csv").exists()

            assert workflow.success

            assert snapshot(workflow).match()

            assert snapshot(workflow.trace.tasks().size()).match()
            assert snapshot(
                "$outputDir/preprocessing/recal_table/test/test.recal.table",
                "$outputDir/reports/mosdepth/test/test.md.mosdepth.global.dist.txt",
                "$outputDir/reports/mosdepth/test/test.md.mosdepth.region.dist.txt",
                "$outputDir/reports/mosdepth/test/test.md.mosdepth.summary.txt",
                "$outputDir/reports/mosdepth/test/test.md.regions.bed.gz",
                "$outputDir/reports/mosdepth/test/test.md.regions.bed.gz.csi",
                "$outputDir/reports/mosdepth/test/test.recal.mosdepth.global.dist.txt",
                "$outputDir/reports/mosdepth/test/test.recal.mosdepth.region.dist.txt",
                "$outputDir/reports/mosdepth/test/test.recal.mosdepth.summary.txt",
                "$outputDir/reports/mosdepth/test/test.recal.regions.bed.gz",
                "$outputDir/reports/mosdepth/test/test.recal.regions.bed.gz.csi",
                "$outputDir/reports/vcftools/strelka/test/test.strelka.variants.FILTER.summary",
                "$outputDir/reports/vcftools/strelka/test/test.strelka.variants.TsTv.count",
                "$outputDir/reports/vcftools/strelka/test/test.strelka.variants.TsTv.qual"
            ).match()
        }
    }
}
