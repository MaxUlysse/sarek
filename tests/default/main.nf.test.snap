{
    "Should run without failures": {
        "content": [
            "test.recal.table:md5,4ac774bf5f1157e77426fd82f5ac0fbe",
            "test.md.mosdepth.global.dist.txt:md5,76fa71922a3f748e507c2364c531dfcb",
            "test.md.mosdepth.region.dist.txt:md5,abc5df85e302b79985627888870882da",
            "test.md.mosdepth.summary.txt:md5,d536456436eb275159b8c6af83213d80",
            "test.md.regions.bed.gz:md5,38fe39894abe62e38f8ac214cba64f2b",
            "test.md.regions.bed.gz.csi:md5,b1c2a861f64e20a94108a6de3b76c582",
            "test.recal.mosdepth.global.dist.txt:md5,76fa71922a3f748e507c2364c531dfcb",
            "test.recal.mosdepth.region.dist.txt:md5,abc5df85e302b79985627888870882da",
            "test.recal.mosdepth.summary.txt:md5,d536456436eb275159b8c6af83213d80",
            "test.recal.regions.bed.gz:md5,38fe39894abe62e38f8ac214cba64f2b",
            "test.recal.regions.bed.gz.csi:md5,b1c2a861f64e20a94108a6de3b76c582",
            "test.strelka.variants.FILTER.summary:md5,dd87f507da7de20d5318841af312493b",
            "test.strelka.variants.TsTv.count:md5,fa27f678965b7cba6a92efcd039f802a",
            "test.strelka.variants.TsTv.qual:md5,bc68ae4e688e9fb772b457069e604883"
        ],
        "timestamp": "2023-01-18T18:39:49+0000"
    }
}